modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -77,58 +77,98 @@ ComputeReturnMappingStress::updateQpStress(RankTwoTensor & strain_increment,
       << std::endl;
   }
 
-  // compute trial stress
-  RankTwoTensor stress_last_iteration = _elasticity_tensor[_qp] * (strain_increment + _elastic_strain_old[_qp]);
-
   RankTwoTensor inelastic_strain_increment, elastic_strain_increment;
   inelastic_strain_increment.zero();
 
-  Real equivalent_delta_stress, first_equivalent_delta_stress;
+  Real l2norm_delta_stress, first_l2norm_delta_stress;
   unsigned int counter = 0;
 
+  std::vector<RankTwoTensor> inelastic_strain_increment_old, inelastic_strain_increment_new;
+  inelastic_strain_increment_old.resize(_models.size());
+  inelastic_strain_increment_new.resize(_models.size());
+
+  RankTwoTensor stress_max, stress_min;
+
+  for (unsigned i_rmm =0; i_rmm < _models.size(); ++i_rmm)
+    inelastic_strain_increment_old[i_rmm].zero();
+
   do
   {
-    elastic_strain_increment = strain_increment;
-    stress_new = _elasticity_tensor[_qp] * (elastic_strain_increment - inelastic_strain_increment + _elastic_strain_old[_qp]);
+    stress_max = RankTwoTensor(-1.0e20, -1.0e20, -1.0e20, -1.0e20, -1.0e20, -1.0e20);
+    stress_min = RankTwoTensor(1.0e20, 1.0e20, 1.0e20, 1.0e20, 1.0e20, 1.0e20);
 
-    for (unsigned i_rmm =0; i_rmm < _models.size(); ++i_rmm)
+    for (unsigned i_rmm = 0; i_rmm < _models.size(); ++i_rmm)
     {
       _models[i_rmm]->setQp(_qp);
+
+      elastic_strain_increment = strain_increment;
+
+      if (i_rmm < _models.size() - 1)
```

Comment:
I assume you're handling the last model differently because it's the plasticity model and the others are creep models. It's in line with how we have been doing this, but I'd like to discuss this more.